### PR TITLE
Using APIs with Fetch

### PR DESCRIPTION
**Pull request by adamclasic**
- Added a form to enter the name of the city.
- Added a background image to display the weather.
- Fixed linter errors

 I look forward to your suggestions.
[Live Demo](http://raw.githack.com/adamclasic/weather-app/feature/dist/index.html)

### CODE REVIEW

**Comment by stratospherique** [comment]
### Project status invalid :red_circle: 

- Friendly reminder: You should use Git Flow in this project.
Please take a step back and watch video [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/).
- **[SUGGESTION]** Create a brand new repo with correct setup and feature branch (with DESCRIPTIVE name) and copy your code there **OR** close this pull request, create a dev branch, make it the default branch and open a new pull request from the feature branch to the new dev branch.

### This Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready! :pray:

**Comment by adamclasic** [comment]
The flow has been corrected.

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

Hi Adam! :wave: 

You need to edit your README file, it has some text from your previous project.

- [ ] The first paragraph has information from another project.

When I run **npm run build** as required from your readme, I was not able to view the application on the browser.

#### ``` Make these changes then submit another form for a code review ```

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

**Comment by adamclasic** [comment]
Build does not start the webserver, it builds an app in dist. You have to run a webserver on the HTML file in `/dist`.

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great progress @team :clap: However, some issues need to be fixed in order to get approval for this project. Check :arrow_down: 

#### Behaviour 'required :warning:
* [x] It has a toggle control to determine if the result will be shown in Celsius or Fahrenheit *(There are two valid solutions: students can make two different requests for the Celsius data and the Fahrenheit data OR make only one request and use math to convert it to the other unit)* ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
* [x] It changes its appearance based on the received data ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app)) 
#### Implementation (required) :warning: 
* [x] Necessary run common commands like `build`, `development` should be in `package.json` inside \`scripts'. Here is a basic example ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/scripts_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))

```javascript
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```

* [x] It uses ES6 syntax ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/es6_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))
  * [x] usage of arrow functions instead of the `function` keyword
  * [x] usage of `import` and `export` instead of `require` or `module.exports`
  * [x] proper usage of `const` and `let` keywords
* [x] It has error handling for `promises` or `async/await` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))
#### General (required)
* [x] Readme should include instructions to run the project locally.
- Kindly list more specific instructions that you are sure will work on any machine.
* [x] No inline rules to bypass linter checks :pray:
### That's it :relieved: When finished kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

File: package.json
Comment (uryelah) [inline package.json]: - [x] [OPTIONAL] Try to also add the 'start' script from the last reviewer example. it will require you to install the webpack-dev-server as a dependency and set its config at the webpack configuration file but it's one of the most convenient and used scripts in all Javascript frameworks.

File: src/index.js
Comment (uryelah) [inline src/index.js]: - [x] You should never use alerts in modern Javascript web applications. A lot of browsers block them and they are nearly impossible to style or to customize. 
[Check this thread about it for more.](https://stackoverflow.com/questions/20359807/why-shouldnt-i-use-alert-in-javascript)
Please show the message in a component at the DOM instead. You can build something that only shows up when there's this error and change its classes/styles and innerText so it looks right.

File: src/scripts/domPopulator.js
Comment (uryelah) [inline src/scripts/domPopulator.js:31]: - [ ] Let's use ES6 more! Use arrow functions instead of the keyword 'function'.
[Check this article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) on more about arrow functions and how to use them.

File: src/scripts/domPopulator.js
Comment (uryelah) [inline src/scripts/domPopulator.js:47]: - [ ] [OPTIONAL] It would be a good idea to save the API base url string at a constants file or similar, that would make it easier to change it in case you want to use another one, avoid typos, etc.

File: src/scripts/getTemp.js
Comment (uryelah) [inline src/scripts/getTemp.js]: - [ ] As mentioned in the comment above, let's use arrow functions in the functions you have here.

File: src/scripts/getTemp.js
Comment (uryelah) [inline src/scripts/getTemp.js]: - [ ] You still need to have error handling even when working with async/await. 
[Please read this article](https://itnext.io/error-handling-with-async-await-in-js-26c3f20bc06a) on try/catch blocks and use them to handle errors for all the fetches you have in this file.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Adam Allalou**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with the improvements you added to your project!

There are still a couple of points you'll need to put more work into though but hang in there and dedicate your heart to the allure of the DOM! 🦾 ⚔️ 

# Required Changes 🏗️ 🐛 💡

- [x] The app appearance should change according to the received data. 
That can be as simple as changing the color of the background depending on the climate/country or even using a different image to each weather it fetches.
You can use element.classlist to remove/add/toggle classes to achieve those changes or even directly change the document style properties.

Get creative with that!


- [x] <REMOVED DUE TO BELONGING TO A DIFFERENT PROJECT @adamclasic sorry for the confusion>

- [x] It's cool you can fetch the data in Celcius or Fahrenheit but the toggle itself should also change the current temperature the user sees on click.
You can do that by making another fetch request if the toggle option is clicked or just convert the temperatures with math. Just make sure the changes are reflected in the DOM.

(Check the comments below too)

-----

 **Kindly fix the pointed out issues and comments below before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

File: src/index.js
Comment (bellom) [inline src/index.js]: kindly remove this comment before merging.

File: src/index.js
Comment (bellom) [inline src/index.js]: This should be a production code, please remove debugging statements. (`console.log`), even if that feature is not fully implemented, users should not see debugging messages.

I trust you could do this before merging.

File: src/index.js
Comment (bellom) [inline src/index.js]: This should be a production code, please remove debugging statements. (`console.log`), even if that feature is not fully implemented, users should not see debugging messages.

I trust you could do this before merging.

**Review by bellom — APPROVED** [review]
## Hi :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by adamclasic** [comment]
I'm sorry to tell you that you have reviewed an old commit. I appreciate the time you put there to look at the code and be honest.

**Pull request merged**
